File: src/app/dashboard/admin/premises/debug_schema.ts

import { createClient } from '@supabase/supabase-js';
import * as dotenv from 'dotenv';
dotenv.config();

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL || '';
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY || '';

if (!supabaseUrl || !supabaseKey) {
  console.error('Missing Supabase environment variables');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function debugSchema() {
  console.log('--- Premises Table Schema ---');
  const { data, error } = await supabase
    .from('premises')
    .select('*')
    .limit(1);

  if (error) {
    console.error('Error fetching from premises:', error.message);
  } else if (data && data.length > 0) {
    console.log('Columns found:', Object.keys(data[0]));
  } else {
    console.log('No data in premises table or table not found.');
  }

  console.log('\n--- Premise Applications Table Schema ---');
  const { data: appData, error: appError } = await supabase
    .from('premise_applications')
    .select('*')
    .limit(1);

  if (appError) {
    console.error('Error fetching from premise_applications:', appError.message);
  } else if (appData && appData.length > 0) {
    console.log('Columns found:', Object.keys(appData[0]));
  } else {
    console.log('No data in premise_applications table.');
  }
}

debugSchema();

File: debug_schema.py

import os
import json
import urllib.request
import urllib.error

# Try to load .env.local manually
env_vars = {}
try:
    with open('.env.local', 'r') as f:
        for line in f:
            if '=' in line and not line.startswith('#'):
                k, v = line.strip().split('=', 1)
                env_vars[k.strip()] = v.strip().strip("'").strip('"')
except:
    pass

url = env_vars.get('NEXT_PUBLIC_SUPABASE_URL')
key = env_vars.get('SUPABASE_SERVICE_ROLE_KEY')

if not url or not key:
    print(f"Missing environment variables: URL={bool(url)}, KEY={bool(key)}")
    exit(1)

def check_table(table_name):
    print(f"--- {table_name} Table Schema ---")
    request_url = f"{url}/rest/v1/{table_name}?limit=1"
    req = urllib.request.Request(request_url)
    req.add_header("apikey", key)
    req.add_header("Authorization", f"Bearer {key}")
    
    try:
        with urllib.request.urlopen(req) as response:
            if response.status == 200:
                data = json.loads(response.read().decode())
                if data:
                    print(f"COLUMNS: {list(data[0].keys())}")
                else:
                    print(f"Table {table_name} is empty")
            else:
                print(f"Error: {response.status}")
    except urllib.error.HTTPError as e:
        print(f"HTTP Error: {e.code} {e.reason} {e.read().decode()}")
    except Exception as e:
        print(f"Error: {str(e)}")

check_table("premises")
check_table("premise_applications")
